src/ringmesh/io/geomodel/io_gm.cpp

```diff
@@ -341,14 +341,25 @@ namespace
             return false;
         }
 
-        void load_file() final
+        std::string generate_folder_name() const
         {
-            const std::string directory_to_unzip{
+            std::string directory_to_unzip{
                 GEO::FileSystem::normalized_path(
                     GEO::FileSystem::dir_name( this->filename_ ) )
                 + "/" + std::to_string(
                             std::hash< std::string >()( this->filename_ ) )
             };
+            index_t unique_id{ 0 };
+            while( GEO::FileSystem::is_directory( directory_to_unzip ) )
+            {
+                directory_to_unzip += std::to_string( unique_id++ );
```

Comment:
I dislike ++. I never know if it is applied or not. Can you put it in another line just above?